datafusion/optimizer/src/eliminate_outer_join.rs

```diff
@@ -39,191 +38,107 @@ impl ReduceOuterJoin {
     }
 }
 
+/// Attempt to eliminate outer joins to inner joins.
+/// for query: select ... from a left join b on ... where b.xx = 100;
+/// if b.xx is null, and b.xx = 100 returns false, filtered those null rows.
+/// Therefore, there is no need to produce null rows for output, we can use
+/// inner join instead of left join.
+///
+/// Generally, an outer join can be eliminated to inner join if equals from where
+/// return false while any inputs are null and columns of those equals are come from
+/// nullable side of outer join.
```

Comment:
We can simplify the docs to just illustrate that " We will replace the outer join by the inner join if the `null` rows in the nullable side will be filtered out.